lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -369,22 +371,30 @@ monitorMetadata
     -> (StakePoolMetadataUrl -> StakePoolMetadataHash -> IO (Either String StakePoolMetadata))
     -> DBLayer IO
     -> IO ()
-monitorMetadata tr gp fetchMetadata DBLayer{..} = forever $ do
-    refs <- atomically (unfetchedPoolMetadataRefs 100)
-
-    successes <- fmap catMaybes $ forM refs $ \(url, hash) -> do
-        traceWith tr $ MsgFetchPoolMetadata hash url
-        fetchMetadata url hash >>= \case
-            Left msg -> Nothing <$ do
-                traceWith tr $ MsgFetchPoolMetadataFailure hash msg
-
-            Right meta -> Just hash <$ do
-                traceWith tr $ MsgFetchPoolMetadataSuccess hash meta
-                atomically $ putPoolMetadata hash meta
-
-    when (null refs || null successes) $ do
-        traceWith tr $ MsgFetchTakeBreak blockFrequency
-        threadDelay blockFrequency
+monitorMetadata tr gp fetchMetadata DBLayer{..} = do
+    -- We keep an in-memory map of peers that fails to respond. Each peer is
+    -- retried three times, after what, they'll simply be ignored. If the server
+    -- is restarted, it'll retry the failed ones.
+    rebels <- newIORef (mempty :: Map StakePoolMetadataHash Int)
+    forever $ do
+        ignoring <- Map.keys . Map.filter (>= 3) <$> readIORef rebels
+        refs <- atomically (unfetchedPoolMetadataRefs 100 ignoring)
+
+        successes <- fmap catMaybes $ forM refs $ \(url, hash) -> do
+            traceWith tr $ MsgFetchPoolMetadata hash url
+            fetchMetadata url hash >>= \case
+                Left msg -> Nothing <$ do
+                    traceWith tr $ MsgFetchPoolMetadataFailure hash msg
+                    modifyIORef rebels (Map.insertWith (+) hash 1)
```

Comment:
What about adding a `MsgWillStartIgnoringMetadata Url` trace for when the new value is `>= 3`?